src/vs/vscode.d.ts

```diff
@@ -1534,12 +1534,31 @@ declare module 'vscode' {
 		 */
 		ignoreFocusOut?: boolean;
 
+		/**
+		 * An optional flag to make the picker multi-select, if true the result is an array of picks.
+		 */
+		multiSelect?: boolean;
+
 		/**
 		 * An optional function that is invoked whenever an item is selected.
 		 */
 		onDidSelectItem?(item: QuickPickItem | string): any;
 	}
 
+	/**
+	 * Represents an item that can be selected in a multi-select quick pick UI.
+	 */
+	export interface MultiSelectQuickPickItem extends QuickPickItem {
+		selected?: boolean;
+	}
+
+	/**
+	 * Options to configure the behavior of the multi-select quick pick UI. (Marker type.)
+	 */
+	export interface MultiSelectQuickPickOptions extends QuickPickOptions {
```

Comment:
I understand why we have to do this but I am not the biggest fan of those type-gymnastics... I wonder if Is there a way to achieve this without adding a new, exported type?

Reply:
`QuickPickOptions & { canSelectMany: true; }` works too. I have added a commit using that, let me know what you think. I'm not a fan of either, but can't think of better alternatives.